src/controllers/hotkeys/HotkeyController.cpp

```diff
@@ -0,0 +1,402 @@
+#include "HotkeyController.hpp"
+#include "common/QLogging.hpp"
+#include "controllers/hotkeys/HotkeyModel.hpp"
+#include "singletons/Settings.hpp"
+
+#include <QShortcut>
+
+namespace chatterino {
+HotkeyController::HotkeyController()
+{
+    this->loadHotkeys();
+}
+
+void HotkeyController::initialize(Settings &settings, Paths &paths)
+{
+}
+
+void HotkeyController::loadHotkeys()
+{
+    auto defaultHotkeysAdded =
+        pajlada::Settings::Setting<std::vector<QString>>::get(
+            "/hotkeys/addedDefaults");
+    auto set = std::set<QString>(defaultHotkeysAdded.begin(),
+                                 defaultHotkeysAdded.end());
+
+    auto keys = pajlada::Settings::SettingManager::getObjectKeys("/hotkeys");
+    this->resetToDefaults(set);
+    pajlada::Settings::Setting<std::vector<QString>>::set(
+        "/hotkeys/addedDefaults", std::vector<QString>(set.begin(), set.end()));
+
+    qCDebug(chatterinoHotkeys) << "Loading hotkeys...";
+    for (const auto &key : keys)
+    {
+        if (key == "addedDefaults")
+        {
+            continue;
+        }
+
+        auto section = "/hotkeys/" + key;
+        auto scopeName =
+            pajlada::Settings::Setting<QString>::get(section + "/scope");
+        auto keySequence =
+            pajlada::Settings::Setting<QString>::get(section + "/keySequence");
+        auto action =
+            pajlada::Settings::Setting<QString>::get(section + "/action");
+        auto arguments = pajlada::Settings::Setting<std::vector<QString>>::get(
+            section + "/arguments");
+        qCDebug(chatterinoHotkeys)
+            << "Hotkey " << scopeName << keySequence << action << arguments;
+
+        if (scopeName.isEmpty() || keySequence.isEmpty() || action.isEmpty())
+        {
+            continue;
+        }
+        HotkeyScope scope;
+        if (scopeName == "tab")
+        {
+            scope = HotkeyScope::Tab;
+        }
+        else if (scopeName == "split")
+        {
+            scope = HotkeyScope::Split;
+        }
+        else if (scopeName == "splitInput")
+        {
+            scope = HotkeyScope::SplitInput;
+        }
+        else if (scopeName == "window")
+        {
+            scope = HotkeyScope::Window;
+        }
+        else if (scopeName == "userCard")
+        {
+            scope = HotkeyScope::UserCard;
+        }
+        else if (scopeName == "settings")
+        {
+            scope = HotkeyScope::Settings;
+        }
+        else
+        {
+            qCDebug(chatterinoHotkeys) << "Unknown scope: " << scopeName;
+            continue;
+        }
```

Comment:
Should make this into an "anonymous namespace" or "static" function in this file